lib/interceptor/cache.js

```diff
@@ -47,27 +47,30 @@ module.exports = (opts = {}) => {
         return dispatch(opts, handler)
       }
 
+      /**
+       * @type {import('../../types/cache-interceptor.d.ts').default.CacheKey}
+       */
+      const cacheKey = makeCacheKey(opts)
+
       // TODO (perf): For small entries support returning a Buffer instead of a stream.
       // Maybe store should return { staleAt, headers, body, etc... } instead of a stream + stream.value?
       // Where body can be a Buffer, string, stream or blob?
-
-      const stream = store.createReadStream(opts)
-      if (!stream) {
+      const result = store.getValueByKey(cacheKey)
+      if (!result) {
         // Request isn't cached
-        return dispatch(opts, new CacheHandler(globalOpts, opts, handler))
+        return dispatch(opts, new CacheHandler(globalOpts, cacheKey, handler))
       }
 
       /**
-       * @param {import('node:stream').Readable} stream
-       * @param {import('../../types/cache-interceptor.d.ts').default.CacheStoreValue} value
+       * @param {import('node:stream').Readable | undefined} stream
+       * @param {import('../../types/cache-interceptor.d.ts').default.CachedResponse} value
        */
       const respondWithCachedValue = (stream, value) => {
-        assert(!stream.destroyed, 'stream should not be destroyed')
-        assert(!stream.readableDidRead, 'stream should not be readableDidRead')
-
+        assert(!stream || !stream.destroyed, 'stream should not be destroyed')
+        assert(!stream || !stream.readableDidRead, 'stream should not be readableDidRead')
         try {
           stream
```

Comment:
Can stream be nully? When?

Reply:
On HEAD requests

Reply:
the asserts already catch it

Reply:
This is broken though like this... onConnect cannot abort the request

Reply:
Wdym?